File: source/SampleNet4/Code/ConversionData.cs

using MasterDevs.ChromeDevTools.Protocol.Chrome.Browser;


namespace Portal_Convert.CdpConverter
{


    public class ConversionData
    {
        private static string s_chromePath;
        public string RemoteDebuggingUrl { get; set; } = "http://localhost";
        public uint RemoteDebuggingPort { get; set; } = 9222;

        public string RemoteDebuggingUri
        {
            get {
                return this.RemoteDebuggingUrl
                    + ":"
                    + this.RemoteDebuggingPort.ToString(System.Globalization.CultureInfo.InvariantCulture);
            }
        }


        public ChromiumActions_t ChromiumActions { get; set; } = ChromiumActions_t.ConvertToImage;

        public int ViewPortWidth { get; set; } = 1024;
        public int ViewPortHeight { get; set; } = 768;

        public double PageWidth { get; set; } = 210;
        public double PageHeight { get; set; } = 297;


        public string Html { get; set; }


        public System.Exception Exception { get; set; }
        public byte[] PngData { get; set; }
        public byte[] PdfData { get; set; }


        public GetVersionCommandResponse Version { get; set; }


        private object s_changeLock = new object();

        public string ChromePath
        {
            get
            {
                if (!string.IsNullOrEmpty(s_chromePath))
                    return s_chromePath;

                lock (s_changeLock)
                {
                    // "~/External/Chromium/x86-" + (System.IntPtr.Size * 8).ToString()
                    if (System.Web.Hosting.HostingEnvironment.IsHosted)
                        s_chromePath = System.Web.Hosting.HostingEnvironment.MapPath(
                            "~/External/Chromium/chrome.exe"
                        );
                    else
                        s_chromePath = MasterDevs.ChromeDevTools.ChromeProcessFactoryHelper.DefaultChromePath;
                } // End lock (s_changeLock) 

                return s_chromePath;
            }
            set
            {
                s_chromePath = value;
            }
        } // End Property ChromePath 


        public ConversionData(string html)
        {
            this.Html = html;
        }

        public ConversionData()
            :this("")
        { }


    } // End Class ConversionData 


}

File: source/SampleNet4VB/Code/ConversionData.vb

Imports MasterDevs.ChromeDevTools.Protocol.Chrome.Browser


Namespace Portal_Convert.CdpConverter


    Public Class ConversionData


        Private Shared s_chromePath As String

        Public Property RemoteDebuggingUrl As String = "http://localhost"
        Public Property RemoteDebuggingPort As UInteger = 9222

        Public ReadOnly Property RemoteDebuggingUri As String
            Get
                Return Me.RemoteDebuggingUrl + ":" + Me.RemoteDebuggingPort.ToString(System.Globalization.CultureInfo.InvariantCulture)
            End Get
        End Property



        Public Property ChromiumActions As ChromiumActions_t = ChromiumActions_t.ConvertToImage
        Public Property ViewPortWidth As Integer = 1024
        Public Property ViewPortHeight As Integer = 768
        Public Property PageWidth As Double = 210
        Public Property PageHeight As Double = 297
        Public Property Html As String
        Public Property Exception As System.Exception
        Public Property PngData As Byte()
        Public Property PdfData As Byte()
        Public Property Version As GetVersionCommandResponse
        Private s_changeLock As Object = New Object()



        Public Property ChromePath As String
            Get
                If Not String.IsNullOrEmpty(s_chromePath) Then Return s_chromePath

                SyncLock s_changeLock

                    If System.Web.Hosting.HostingEnvironment.IsHosted Then
                        s_chromePath = System.Web.Hosting.HostingEnvironment.MapPath("~/External/Chromium/chrome.exe")
                    Else
                        s_chromePath = MasterDevs.ChromeDevTools.ChromeProcessFactoryHelper.DefaultChromePath
                    End If
                End SyncLock

                Return s_chromePath
            End Get
            Set(ByVal value As String)
                s_chromePath = value
            End Set
        End Property


        Public Sub New(ByVal html As String)
            Me.Html = html
        End Sub


        Public Sub New()
            Me.New("")
        End Sub


    End Class


End Namespace
